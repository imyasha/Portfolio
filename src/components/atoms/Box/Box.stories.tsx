import { Box } from './Box';

export default {
	title: 'atoms/Box',
	component: Box,
};

export const BoxStory = ({ children }: any) => <Box>{children}</Box>;

BoxStory.storyName = 'Box';
BoxStory.args = {
	children: 'Hello Box',
};
