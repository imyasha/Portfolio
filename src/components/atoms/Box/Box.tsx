import { FC, ReactNode } from 'react';

interface BoxProps {
	children: ReactNode;
}

const Box: FC<BoxProps> = ({ children }) => {
	return (
		<div className="md:w-10/12 px-8 py-6 rounded-3xl glass-light dark:glass-dark mb-6 drop-shadow-lg hover:drop-shadow-2xl transition-all">
			{children}
		</div>
	);
};

export { Box };
