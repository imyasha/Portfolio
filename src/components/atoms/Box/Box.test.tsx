import { Box } from '../Box';
import { render } from '@testing-library/react';

describe('Box', () => {
	it('renders correctly', () => {
		const { container } = render(<Box>Hello Box</Box>);
		expect(container).toMatchSnapshot();
	});
});
