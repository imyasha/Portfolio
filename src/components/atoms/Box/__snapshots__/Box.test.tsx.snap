// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders correctly 1`] = `
<div>
  <div
    class="md:w-10/12 px-8 py-6 rounded-3xl glass-light dark:glass-dark mb-6 drop-shadow-lg hover:drop-shadow-2xl transition-all"
  >
    Hello Box
  </div>
</div>
`;
